src/controller/java/src/chip/onboardingpayload/OnboardingPayload.kt

```diff
@@ -182,8 +182,8 @@ class OnboardingPayload(
       DiscoveryCapability.SOFT_AP
     )
 
-    // If discoveryCapabilities is empty or discoveryCapabilities contains values outside of allValid
-    if (discoveryCapabilities.isEmpty() || discoveryCapabilities.any { it !in allValid }) {
+    // If discoveryCapabilities is not empty and discoveryCapabilities contains values outside of allValid
+    if (!discoveryCapabilities.isEmpty() && discoveryCapabilities.any { it !in allValid }) {
```

Comment:
Can tests be added. This should have failed a test before, and now passing

Reply:
We don't have unit test framework for kotlin QRCode yet, I am working on it right now and will post the PR soon. It will cover this check and also integrated in CI